src/nouislider.ts

```diff
@@ -168,6 +169,19 @@ export class NouisliderComponent implements ControlValueAccessor, OnInit {
     });
   }
 
+  ngOnChanges(changes: any) {
+    if (this.slider && (changes.min || changes.max)) {
+      setTimeout(() => {
+        this.slider.updateOptions({
+          range: {
+            min: this.min,
+            max: this.max
```

Comment:
The cdRef.detectChanges() will not work in the nouislider.ts. The exception comes from the app component in the demo. If you apply cdRef.detectChanges() in the app component after where you assign new values to min and max then it will work. Also I think that only changing min and max is not sufficient enough since nouislider gives the opportunity to make a range object as follows {min:10, '20%':20,max:30}. Min and max will simply not work in this situation. I would suggest to make deep changes in the component and actually remove min and max inputs and add a range input which is actually an object. Then when we want to change the min and max we will just assign a new value to our current  range property.